external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaRelationSuite.scala

```diff
@@ -228,19 +233,76 @@ class KafkaRelationSuite extends QueryTest with SharedSQLContext with KafkaTest
     }, topic, 0 to 19)
   }
 
+  test("no matched offset for timestamp - startingOffsets") {
```

Comment:
I add this test only for batch test since also adding this to streaming test feels redundant, but I can add it to streaming test if we want to.